ddtrace/settings.py

```diff
@@ -103,6 +104,55 @@ def __repr__(self):
         return '{}.{}({})'.format(cls.__module__, cls.__name__, integrations)
 
 
+class IntegrationConfigItem(object):
+    """
+    A wrapper for items stored in an IntegrationConfig which is responsible for
+    determining what the value of the item stored in the config should be.
+
+    The precedence for determining the value for a config item is:
+
+    [high]
+        1) user-overridden value (indicated via `update_value`)
+        2) environment variable  (of the form DD_{INTEGRATION}_{KEY})
+        3) default value         (specified in the initializer)
+    [low]
+
+    TODO:
+        - env_var_names param to allow custom environment variables to be checked
+        - docstring param to allow the item to be described by the integration
+        - provide a method for the item to be reset back to the default after a
```

Comment:
This might be the only item in the list that we may not want to tackle.

Reply:
yeah, people won't likely try to revert their configuration, at least at runtime, and they can just do `config.redis.service_name = 'new-value'`